### PR TITLE
Propagate exit codes correctly in Windows scripts

### LINKED ISSUE #15428: Gradle Windows scripts don't propagate exit codes correctly
Gradle scripts on Windows don't care what exit code an underlying app has returned - as long as it's non-zero, they will return `1`. This behaviour is inconsistent with how these scripts work on Unix-based operating systems.

### Expected Behavior
An exact exit code is returned.

### Current Behavior
`0` on success, `1` on anything else (regardless of an underlying exit code). Both `gradlew.bat` and scripts generated by `installDist` task behave like this.

### Context
I was writing command-line scripts to complement the ones created by `installDist` in order to pass command-line arguments more naturally (i.e. not by `gradlew run --args='--something --verbose -u'`) when I've noticed this behaviour. Ideally I would like to capture exact exit codes out-of-the-box.

### Steps to Reproduce 
I've illustrated an example here: https://github.com/mataha/gradle-windows-exit-code

```batchfile
git clone https://github.com/mataha/gradle-windows-exit-code
cd gradle-windows-exit-code
:: Expectedly should print: `Exit code: 2`
run 2
```

Output: `Exit code: 1`

### PR DESCRIPTION
**Pull request by mataha**
Fixes #15428

### Context

See #15428.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: gradlew.bat [hunk 1]
```diff
@@ -80,8 +80,8 @@ if "%ERRORLEVEL%"=="0" goto mainEnd
 :fail
 rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
 rem the _cmd.exe /c_ return code!
-if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
-exit /b 1
+if  not "" == "%GRADLE_EXIT_CONSOLE%" exit %ERRORLEVEL%
```
Comment (sschuberth) [inline gradlew.bat]: Nit: Mind getting rid of the double-space after `if` while at it? Similar below.

Comment (sschuberth) [reply gradlew.bat]: Sorry, I just realized you're already doing this in the third commit 👍🏻

**Comment by sschuberth** [comment]
LGTM, IMO this is a low-hanging fruit that should get merged soon @big-guy.

**Comment by mataha** [comment]
Given that I have deleted my fork of the repo a few months ago, is it even possible to merge this right now?

**Comment by sschuberth** [comment]
It should be, yes. As you can see the diff is still visible, and that's because PRs do not directly refer to the branch in your fork, but to a special "branch" in the upstream repo in the "pull" ref namespace. Also see https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally#modifying-an-inactive-pull-request-locally.

**Comment by sschuberth** [comment]
@jjohannes can we please reopen this and move it forward? It seems to be good to merge to me.

**Comment by mataha** [comment]
Is there a problem with this PR I don't see, or is it just a very low priority?

**Comment by sschuberth** [comment]
Gradle team, please take the time to review this; it's not much code.

**Comment by sschuberth** [comment]
Again, @jjohannes, @big-guy, anyone, please have a look at this!

**Comment by sschuberth** [comment]
@jjohannes, @big-guy, I'm not getting tired of pinging you 😉

**Comment by sschuberth** [comment]
> If you intend to work on this pull request

IMO there's no need to work further on this PR, it's ready to get merged.

**Comment by ljacomet** [comment]
@bot-gradle test this

**Comment by ljacomet** [comment]
Closing in favor of #19865 

@sschuberth Thank you for your persistence, sorry it took so long.
@mataha Thanks for the contribution!

**Pull request closed without merging**